src/invidious/routes/misc.cr

```diff
@@ -43,4 +43,14 @@ module Invidious::Routes::Misc
     instance_url = fetch_random_instance
     env.redirect "https://#{instance_url}#{referer}"
   end
+
+  def self.confirm_leave(env)
+    locale = env.get("preferences").as(Preferences).locale
+    
+    link = env.params.query["link"]?
```

Comment:
Change 1: the link parameter is mandatory
Change 2: HTML escape the data, otherwise this introduces an XSS vulnerability

NB: you will need to add `require "html"` at the top of the file to make it work.

```suggestion
    link = HTML.escape(env.params.query["link"])
```